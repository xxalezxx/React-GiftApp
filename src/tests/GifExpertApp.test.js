import { shallow } from "enzyme"
import React from "react"

import GifExpertApp from "../GifExpertApp"

describe("GifExpertApp testing - Normal function of it", () => {
  test("Should match with the snapshot", () => {
    const wrapper = shallow(<GifExpertApp />)
    expect(wrapper).toMatchSnapshot()
  })

  test("Should have 2 elements because I send it array with 2 elements", () => {
    const categories = ["Batman", "Peter"]
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />)
    expect(wrapper.find("GifGrid").length).toBe(categories.length)
  })
})
