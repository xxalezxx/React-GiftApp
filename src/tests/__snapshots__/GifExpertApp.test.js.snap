// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp testing - Normal function of it Should match with the snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
